catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/AbstractGmdTransformer.java

```diff
@@ -36,20 +38,33 @@
 import org.codice.ddf.spatial.ogc.csw.catalog.common.GmdConstants;
 import org.codice.ddf.spatial.ogc.csw.catalog.converter.CswRecordConverter;
 
-public class AbstractGmdTransformer implements MetacardTransformer {
+public class AbstractGmdTransformer implements MetacardTransformer, QueryResponseTransformer {
 
   public static final String GML_PREFIX = "gml:";
 
   public static final String GCO_PREFIX = GmdConstants.GCO_PREFIX + ":";
 
   private Supplier<Converter> converterSupplier;
 
+  protected static final String TRANSFORM_EXCEPTION_MSG =
+      "Unable to transform from GMD Metadata to Metacard";
+
   /** @param converterSupplier must be non-null */
   public AbstractGmdTransformer(Supplier<Converter> converterSupplier) {
     notNull(converterSupplier, "converterSupplier must be non-null");
     this.converterSupplier = converterSupplier;
   }
 
+  @Override
+  public BinaryContent transform(SourceResponse sourceResponse, Map<String, Serializable> map)
+      throws CatalogTransformerException {
+    if (sourceResponse.getResults() != null && sourceResponse.getResults().size() == 1) {
```

Comment:
the GMD Schema itself doesn't appear to support multiple metadata records in a single doc, but a possible improvement here would be to create a zip of all the results in the GMD XML format.